Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,206 @@
+                this.postImpressionService.AddPostImpressions(nullPostImpression);
+
+            //then
+            await Assert.ThrowsAsync<PostImpressionValidationException>(() =>
+                addPostImpressionTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedPostImpressionValidationException))),
+                        Times.Once);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnAddIfPostImpressionIsInvalidAndLogItAsync()
+        {
+            //given
+            Guid invalidGuid = Guid.Empty;
+
+            var invalidPostImpression = new PostImpression
+            {
+                PostId = invalidGuid,
+                ProfileId = invalidGuid
+            };
+
+            var invalidPostImpressionException =
+                new InvalidPostImpressionException();
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.PostId),
+                values: "Id is required.");
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.Post),
+                values: "Post is required.");
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.ProfileId),
+                values: "Id is required.");
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.Profile),
+                values: "Profile is required.");
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.CreatedDate),
+                values: "Date is required.");
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.UpdatedDate),
+                values: "Date is required.");
+
+            var expectedPostImpressionValidationException =
+                new PostImpressionValidationException(invalidPostImpressionException);
+
+            //when
+            ValueTask<PostImpression> addPostImpressionTask =
+                this.postImpressionService.AddPostImpressions(invalidPostImpression);
+
+            //then
+            await Assert.ThrowsAsync<PostImpressionValidationException>(() =>
+                addPostImpressionTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedPostImpressionValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertPostImpressionAsync(invalidPostImpression),
+                    Times.Never);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
```

Comment:
Do the same for rest

Reply:
I was able to add the dateTimeBroker for the other tests in the commit  CODERUB: Created The DateTimeBrokerMock For Tests In Validation